SalesforceCommerceCloud/pwa-kit
# Fix NPM dependencies

# Description


The PR uses NPM 8 to re-generate the lock files using `lockfileVersion:2` which is backwards compatible with `lockfileVersion:1` solving the NPM warning `npm WARN old lockfile`.


Regenerating lock files using NPM 8 uncovered a few problems with the dependencies resolution in our package.json files.

These are the changes done to successfully re-generate lock files using NPM 7 or 8:


#### Package `template-retail-react-app`
- Add `@chakra-ui/system` v1 to prevent downloading an incompatible v2.
- Add `@testing-library/dom` to `devDependecies` since is used by some tests: 
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/d64503d3d56506eee325087fd1bc7219b027aa22/packages/template-retail-react-app/app/components/icons/index.test.js#L8

#### Package `pwa-kit-react-sdk`
- Remove `enzyme-adapter-react-16`. Since requires react@16 and we currently use react@17.

#### Generator script
After re-generating the package-lock.json files, the generator script was stuck waiting for the local NPM Verdaccio.
A recent change made in the integration hooks branch on a [PR also regenerating lock files](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/994/files#diff-20c8675dfa83a40be8f874f30eef9f0d57416a87df671d0d91c527a3ee413f32) seems to solve that problem here as well.

-----

The PR also fixes a Warning when pushing a bundle to MRT using Node 16.
The change is backwards compatible with Node 14.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`


# How to Test-Drive This PR

Verify we can re-generate lock files using NPM 8.
1. Checkout this branch `fix-npm-peer-dependencies`. Remove node_modules folders and package-lock.json files
```
git clean -dxf && find . -name 'package-lock.json' -type f -delete
```
2. Switch to NPM 7 or NPM 8
```
nvm use 16.19
```
3. Install dependencies
```
npm install
```
4. Confirm lock files are being generated.

-----

Verify we don't see a warning when pushing a bundle to MRT using Node 16.
 1. Using Node 16, run the `npm run push` command to push a bundle to an MRT environment. 
 2. Verify there're no Node 16 warnings in the terminal.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
